src/codegen/mod.rs

```diff
@@ -4411,6 +4616,52 @@ mod utils {
         }
     }
 
+    pub fn type_id_to_rust_type(
```

Comment:
Does this need to be public? Also, I think the name of this function should have something to do with arguments. The conversions done here are only valid for arguments.

Reply:
If I remove the "pub" it does not compile because try_codegen_operator calls it.
I changed the name to argument_type_id_to_rust_type in the next commit.